src/main/java/org/wirvsvirus/locoronando/location/GeocodingServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.wirvsvirus.locoronando.location;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.wirvsvirus.locoronando.dealer.Address;
+
+@Service
+public class GeocodingServiceImpl implements GeocodingService {
+
+	private static final String ZIPCODE_SEARCH_STRING = "%s Germany";
+	@Value("${geocode.apikey}")
+	private String apikey;
+	@Value("${geocode.url}")
+	private String url;
+
+	@Autowired
+	private RestTemplate restTemplate;
+
+	/**
+	 * Gets geo coordinates from a description.
+	 *
+	 * @param description Free text description of a place.
+	 * @return geo coordinates
+	 */
+	private Optional<Geocode> getGeocodeFromDescription(String description) {
```

Comment:
Change the method name to find, because it is not guaranteed that you will find the object.

Reply:
done